src/plugins/data/server/index_patterns/deprications/scripted_fields.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import {
+  CoreSetup,
+  DeprecationsDetails,
+  GetDeprecationsContext,
+  RegisterDeprecationsConfig,
+} from 'kibana/server';
+import { IndexPatternAttributes } from '../../../common';
+
+type IndexPatternAttributesWithFields = Pick<IndexPatternAttributes, 'title' | 'fields'>;
+
+export const createScriptedFieldsDeprecationsConfig: (
+  core: CoreSetup
+) => RegisterDeprecationsConfig = (core: CoreSetup) => ({
+  getDeprecations: async (context: GetDeprecationsContext): Promise<DeprecationsDetails[]> => {
+    const finder = context.savedObjectsClient.createPointInTimeFinder<IndexPatternAttributesWithFields>(
+      {
+        type: 'index-pattern',
+        perPage: 100,
+        fields: ['title', 'fields'],
+      }
+    );
+
+    let indexPatternsWithScriptedFields: IndexPatternAttributesWithFields[] = [];
+    for await (const response of finder.find()) {
```

Comment:
We fetch 100 index patterns per page using the new finder API. See for more context: https://github.com/elastic/kibana/issues/93770

I intentionally do not reuse index pattern service to optimize pagination here and to fetch only `fields` and `title`

Reply:
I was thinking about whether it would make sense to make something like this officially supported by the index pattern service but the fact that this is used to look for scripted fields seems to make this a one-off.